rasa_core/policies/ensemble.py

```diff
@@ -331,10 +331,24 @@ def probabilities_using_best_policy(self,
                                         tracker: DialogueStateTracker,
                                         domain: Domain
                                         ) -> Tuple[List[float], Text]:
+
+        fallback_idx_policy = [(i, p) for i, p in enumerate(self.policies)
+                               if isinstance(p, FallbackPolicy)]
+
+        fallback_idx, fallback_policy = -1, None
+        if fallback_idx_policy:
+            fallback_idx, fallback_policy = fallback_idx_policy[0]
+            # evaluate the fallback policy last, so that we are able to
+            # determine that there is no other policy with higher action
+            # prediction confidence than the core threshold
+            self.policies.append(self.policies.pop(fallback_idx))
+            fallback_idx = self.policies.index(fallback_policy)
```

Comment:
I worry that changing the index of the fallback policy might mess things up when at the end of this method we return the name of the best policy (which includes the index)